src/components/widgets/QRCode.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import "../../styles/widget.css"
+import qrCode from 'qrcode-npm'
+
+class QRCode extends Component {
+
+  constructor( props ) {
+    super( props )
+    this.generateQRCode = this.generateQRCode.bind( this )
+  }
+
+  generateQRCode( url ) {
+    const qr = qrCode.qrcode( 4, 'M' )
+
+    qr.addData( url )
+    qr.make()
+
+    return qr.createImgTag( 4 )
+  }
+
+  render() {
+    const { specification } = this.props
+    const imageTag = this.generateQRCode( specification.url )
+    const src = imageTag.replace( '" width="164" height="164"/>', '' ).replace( '<img src="', '' )
```

Comment:
Since we have to do additional processing to get the `qrcode-npm` functionality to work for us, I'd be tempted to wrap it in its own component...